### PR TITLE
chore(github): add PR templates for default and release PRs

### PR DESCRIPTION
**Pull request by SanyaKochhar**
* Default PR template similar to OSS PR template, auto populates when you open a PR
* Release PR template can be used by adding `?expand=1&template=release_pull_request_template.md` to the PR URL during creation or copying the raw template

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    1. Updated the chart version, app version and dependency version in charts/osm-arc/Chart.yaml    
+    
+    2. Made applicable updates to the osm-arc values.yaml if any were made in the upstream OSM chart
+    <!--
+    In upstream, compare between the latest release and the previous release to check if anything has changed in the OSS values.yaml e.g: https://github.com/openservicemesh/osm/compare/v0.6.1...v0.7.0-rc.1
+    Check for variable name changes, removed variables, variables that need to be overridden, etc. and make applicable changes in the osm-arc chart.    
+    -->   
+
+
+Is this a release candidate?
```
Comment (ksubrmnn) [inline .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md:21]: Is there any action item for this? Can you give more context as to why this question is being asked and answered?

Comment (SanyaKochhar) [reply .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md]: nope, it was a relic from the upstream checklist. Removed now

**Comment by nshankar13** [comment]
For the release PRs, I would add the following notes: 

- I have verified that the images for the new versions of osm-controller, init, and osm-injector are in MCR. 
- This PR has received 3 approvals before merging.

**Review by ksubrmnn — DISMISSED** [review]

**Review by nshankar13 — APPROVED** [review]

**Review by ksubrmnn — APPROVED** [review]

**Pull request merged**